src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -812,6 +831,38 @@ private static bool CanReplace(ISymbol symbol)
             return memberAccess.CanReplaceWithReducedName(replacementNode, semanticModel, cancellationToken);
         }
 
+        /// <summary>
+        /// Determines if <paramref name="speculativeSymbol"/> contains a superset of the symbols in
+        /// <paramref name="actualSymbol"/>.
+        /// </summary>
+        private static bool IsReplacementCandidate(SymbolInfo actualSymbol, SymbolInfo speculativeSymbol)
```

Comment:
1. I believe it's possible that `actualSymbol` contains multiple symbols even in valid code (e.g. `nameof` applied to a method group)
1. The lookup doesn't perform overload resolution for invocations, so it's possible that the speculative symbols to contain multiple possibilities that still resolve to the same symbol in the end